### PR TITLE
Solution to Issue 7 and 9

### PR DESCRIPTION
**Pull request by allen-woods**
Refactored math functions and permitted calculation of common dimensions between vectors. Improved accuracy of floating point calculations using toFixed.

### CODE REVIEW

File: src/vector.js [hunk 1]
```diff
@@ -346,18 +295,25 @@ class Vector {
 
       if (a.length === b.length && a.length === 3) {
         let params = [
-          a[1] * b[2] - a[2] * b[1],
```
Comment (spikeburton) [inline src/vector.js]: This was cleaner and simpler

File: src/vector.js [hunk 2]
```diff
@@ -346,18 +295,25 @@ class Vector {
...
-          a[2] * b[0] - a[0] * b[2],
-          a[0] * b[1] - a[1] * b[0]
+          Vector.vectorSub(
```
Comment (spikeburton) [inline src/vector.js]: Let's remove this

File: src/vector.js [hunk 3]
```diff
@@ -220,70 +222,34 @@ class Vector {
...
       result = new Vector(...params);
     } catch (error) {
       console.log(`ERROR: ${error}`);
-    } finally {
-      // do nothing
     }
     return result;
   }
 
-  static div(scalar, vectorObj) {
```
Comment (spikeburton) [inline src/vector.js]: Nit picky but I preferred the scalar as the first argument as was previously implemented. This also introduces a problem of code not being backwards compatible (big thing to keep in mind)

File: src/vector.js [hunk 4]
```diff
@@ -135,69 +133,75 @@ class Vector {
...
-          for (let c = 0; c < a.length; c++) {
-            // Store the result of the operation performed inside opFunc.
+        let m = false;
+        let n = false;
 
-            // We need to programmatically define the init value because
-            // multiplication requires 1 as the init value, not zero.
-            results.push(
-              [a[c], b[c]].reduce(opFuncArray[0].f, opFuncArray[0].i)
+        if (a.length === 1 || b.length === 1) {
```
Comment (spikeburton) [inline src/vector.js]: 👍good stuff here

File: src/vector.js [hunk 5]
```diff
@@ -135,69 +133,75 @@ class Vector {
 
   /* Class Methods */
 
-  // Array reduce callbacks
-  static vectorAdd(acc, cur) {
-    return acc + cur;
+  static vectorAdd(a, b) {
```
Comment (spikeburton) [inline src/vector.js]: 👍 This is great

File: src/vector.js (hunk 1 above)
Comment (allen-woods) [reply src/vector.js]: The previous implementation introduces rounding errors.

File: src/vector.js (hunk 2 above)
Comment (allen-woods) [reply src/vector.js]: Removal of this change will result in inaccurate calculations.

File: src/vector.js (hunk 3 above)
Comment (allen-woods) [reply src/vector.js]: Argument order rolled back, currently pushed to this branch.

File: src/vector.js (hunk 4 above)
Comment (allen-woods) [reply src/vector.js]: Sweet! I love a good logic operator.

**Pull request merged**